### PR TITLE
[Atk] Fix Atk.Object leaks.

### PR DESCRIPTION
**Pull request by Therzok**

### CODE REVIEW

**Comment by alanmcgovern** [comment]
Just copying this comment from the old PR:
```
I'm a little worried by the Atk change because there's a method called ref_child and it no longer refs the child. We've definitely broken the semantics behind this method. It looks like native code is expecting a native ref to be added, but with this change we are not doing that. This method is now a no-op for the case where the managed wrapper already exists, and for the case where the managed wrapper does not exist we add a ref, but it's a ref owned by managed, not by native.
```

File: atk/Object.custom [hunk 1]
```diff
@@ -88,8 +89,10 @@
 		}
 
 		[GLib.DefaultSignalHandler (Type=typeof(Atk.Object), ConnectionMethod="OverrideRefChild")]
-		protected virtual Atk.Object OnRefChild (int i) {
-			return null;
+		protected virtual Atk.Object OnRefChild (int i)
+		{
+			IntPtr raw = atksharp_object_base_ref_child (Handle, i);
+			return GLib.Object.GetObject (raw, true) as Atk.Object;
```
Comment (alanmcgovern) [inline atk/Object.custom]: Is this one of the methods which *may not* be implemented in the base class, which means we could get `IntPtr.Zero` as raw? If so, is it safe to call GetObject with a null pointer?

**Comment by alanmcgovern** [comment]
The changes seem reasonable to me but i haven't tested em. I'll leave it to Iain's best judgement ;)

File: atk/Object.custom (hunk 1 above)
Comment (Therzok) [inline atk/Object.custom]:
```
public static Object GetObject(IntPtr o, bool owned_ref)
		{
			if (o == IntPtr.Zero)
				return null;
```

**Comment by Therzok** [comment]
Re-review please, I added more things onto it that should fix bindings for Atk#

**Comment by iainx** [comment]
Will the objectfactory  changes mean that native code can create AtkObjects that are created in managed code?

**Comment by Therzok** [comment]
Yes

**Comment by iainx** [comment]
yet :) that will make the custom accessibility stuff much easier

**Comment by Therzok** [comment]
At least, I think that's how it's going to work, the binding is now fully implemented - with override and base calls, so it _should_ work just fine.

**Comment by alanmcgovern** [comment]
Should we merge this if everyone is happy?

**Comment by Therzok** [comment]
Yup, let's go ahead

**Pull request merged**